Add YouTube links to docs (#7156)
marimo-team/marimo: docs/guides/reusing_functions.md

your notebook code reusable, testable, and easier to edit in text editors
of your choice.

> > Our YouTube channel also features a full video tutorial on this feature, which you can view [here](https://youtu.be/4AFcgd-s3Fg).

Reviewer: ```suggestion
> Prefer learning by video? Watch our [tutorial on YouTube](https://youtu.be/4AFcgd-s3Fg).
```
